calico-vpp-agent/watchers/net_watcher.go

```diff
@@ -248,74 +255,30 @@ func (w *NetWatcher) OnNetDeleted(netName string) error {
 	return nil
 }
 
-func getNetworkVrfName(networkName string, suffix string) string {
-	return fmt.Sprintf("pod-%s-table-%s", networkName, suffix)
-}
-
-func (w *NetWatcher) CreateVRFsForNet(networkName string, networkVni uint32, netRange string) (netDef *NetworkDefinition, err error) {
+func (w *NetWatcher) CreateNetwork(networkName string, networkVni uint32, netRange string, phyNet string) (netDef *NetworkDefinition, err error) {
 	/* Create and Setup the per-network VRF */
-	var tables [2]uint32
 	if _, ok := w.networkDefinitions[networkName]; ok {
 		return w.networkDefinitions[networkName], nil
 	}
 	w.log.Infof("adding network %s", networkName)
-	swIfIndex, err := w.vpp.CreateLoopback(common.ContainerSideMacAddress)
-	if err != nil {
-		return nil, errors.Wrapf(err, "Error creating loopback for network")
-	}
-	w.log.Infof("loopback (sw %+v) added for network %s", swIfIndex, networkName)
-
-	for idx, ipFamily := range vpplink.IpFamilies {
-		vrfName := getNetworkVrfName(networkName, ipFamily.Str)
-		vrfId, err := w.vpp.AllocateVRF(ipFamily.IsIp6, vrfName)
-		w.log.Infof("Allocated %s VRF ID:%d", ipFamily.Str, vrfId)
-		if err != nil {
-			return nil, errors.Wrapf(err, "error allocating VRF %s", ipFamily.Str)
-		}
-		tables[idx] = vrfId
-	}
+	vrfID := common.VppManagerInfo.PhysicalNets[phyNet].VrfId
+	podVrfID := common.VppManagerInfo.PhysicalNets[phyNet].PodVrfId
+	swIfIndex := common.VppManagerInfo.PhysicalNets[phyNet].SwIfIndex
```

Comment:
Do we really need to store `vrfID` `podVrfID` and `swIfIndex` in the `NetworkDefinition` ?
iiuc the only place this is used is in `vxlan.go` and we can directly do the fetch there in `common.VppManagerInfo.PhysicalNets[phyNet]`

Reply:
Correct for swifindex, it is only used in vxlan connectivity add to set the vxlan unnumbered, but for vrfs, they are used in calico vpp cni as well to create pods routes in vpp in multinet.